core/commands/checkout.py

```diff
@@ -129,3 +130,21 @@ def run(self):
         if self.file_path:
             self.checkout_file(self.file_path)
             self.window.status_message("Successfully checked out {} from head.".format(self.file_path))
+            util.view.refresh_gitsavvy_interfaces(self.window, interface_reset_cursor=True)
+
+
+class GsCheckoutCurrentFileAtCommitCommand(LogMixin, WindowCommand, GitCommand):
+
+    """
+    Reset the current active file to a given commit.
+    """
+
+    def run(self):
+        if self.file_path:
+            super().run(file_path=self.file_path)
+
+    def do_action(self, commit_hash, **kwargs):
+        if commit_hash:
+            self.checkout_ref(commit_hash, self.file_path)
+            self.window.status_message("Successfully checked out {} from head.".format(self.file_path))
```

Comment:
Should we maybe mention that we checked out from {commit_hash} rather than from "head" ?

Reply:
You are right.